tests/test_downloader/test_executors/test_base.py

```diff
@@ -0,0 +1,170 @@
+
+import re
+from unittest.mock import Mock, call
+
+from django.test import TestCase
+import pytest
+
+from downloader.models import DownloadRequest
+from downloader.executors.athena import AthenaExecutor
+from tests.factory import EntityFactory
+
+
+ef = EntityFactory()
+
+
+def normalize_query(q):
+    return re.sub(r'\s+', ' ', q).strip()
+
+
+class BaseExecutorTestCase(TestCase):
+
+    @pytest.fixture(autouse=True)
+    def initfixtures(self, mocker):
+        self.mocker = mocker
+
+        self.mocker.patch('downloader.executors.athena.sleep')
```

Comment:
why do you mock `athena` context here since we're in the base?

Reply:
Yeah... I was considering which scenario to follow:
1. Artificially try to stick/limit to BaseExecutor context here
2. Use 'in the wild' scenario - as these methods will be used in practice.
To start, I've followed the second one as the simpler approach.

Now I see your next comment with nice consensus between above - will work on that!

Reply:
actually...

> Simple is better than complex